docs/installation.rst

```diff
@@ -77,9 +77,14 @@ Execute the following commands to install Transformer Engine from source:
 
 .. code-block:: bash
 
-  git clone https://github.com/NVIDIA/TransformerEngine.git    # Clone the repository/fork.
-  cd TransformerEngine                                         # Enter TE directory.
-  git checkout stable                                          # Checkout the correct branch.
-  git submodule init; git submodule update                     # Checkout the submodules.
-  export NVTE_FRAMEWORK=pytorch                                # Optionally set the framework.
-  pip install .                                                # Build and install.
+  git clone --recursive https://github.com/NVIDIA/TransformerEngine.git    # Clone the repository/fork and checkout all submodules recursively.
+  cd TransformerEngine                                                     # Enter TE directory.
+  git checkout stable                                                      # Checkout the correct branch.
+  export NVTE_FRAMEWORK=pytorch                                            # Optionally set the framework.
+  pip install .                                                            # Build and install
+
+For already cloned repos, run the following command in TE directory:
+
+.. code-block:: bash
+
+  git submodule update --init --recursive                                   # Checkout all submodules recursively.
```

Comment:
This seems a bit overkill to me.

Reply:
It kind of is, but the same error has popped up on several occasions now. So it's better to be future safe and comprehensive.